Public/Src/Utilities/Utilities/Tracing/BinaryLogReader.cs

```diff
@@ -224,7 +224,11 @@ public EventReadResult ReadEvent()
                         Contract.Assert(LogStream.Position <= (position + header.EventPayloadSize), "Event handler read beyond the event payload");
                     }
 
-                    m_logStreamReader.ReadBytes((int)(m_nextReadPosition.Value - LogStream.Position));
+                    if (LogStream.Position != m_nextReadPosition.Value)
+                    {
+                        LogStream.Seek(m_nextReadPosition.Value, SeekOrigin.Begin);
+                    }

```

Comment:
I'm not sure that I understand the purpose of this. could you pls explain?

Reply:
Ahh this is some magic that Lance wrote so that if we are not reading over some event data, instead of having to compute the result of (int)(m_nextReadPosition.Value - LogStream.Position), and storing in memory, we just seek forward. It reduced the memory usage by like 30% when we ran the profiler on Thursday for analyzers like EventStatsAnalyzer

---
In reply to: [308912705](https://github.com/microsoft/BuildXL/pull/671#discussion_r308912705) [](ancestors = 308912705)